src/Operation/ListCollectionNames.php

```diff
@@ -67,15 +68,16 @@ public function __construct(string $databaseName, array $options = [])
     /**
      * Execute the operation.
      *
-     * @see Executable::execute()
      * @return Iterator
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
     public function execute(Server $server): Iterator
     {
-        return new CallbackIterator(
-            $this->listCollections->execute($server),
-            fn (array $collectionInfo): string => (string) $collectionInfo['name'],
+        return new CachingIterator(
```

Comment:
Should the return type of this method be typed as an iterator of strings?

Likewise for `ListDatabaseNames::execute()`. I'm not sure what the corresponding helper methods use, but I'll leave that for you to double-check.

Reply:
Changed this for `listCollections` and `listDatabases`.